radixdlt/src/integration/java/com/radixdlt/integration/distributed/deterministic/DeterministicRandomlySyncedLedgerModule.java

```diff
@@ -42,19 +42,20 @@ public DeterministicRandomlySyncedLedgerModule(Random random) {
 		this.random = random;
 	}
 
-	@Provides
-	private SyncRequestSender syncRequestSender() {
-		return request -> { };
+	@Override
+	public void configure() {
+		bind(NextCommandGenerator.class).toInstance((view, aids) -> null);
+		bind(SyncRequestSender.class).toInstance(req -> { });
 	}
 
+	@Provides
 	@Singleton
-	@ProvidesIntoSet
 	Ledger syncedExecutor(CommittedStateSyncSender committedStateSyncSender) {
 		return new Ledger() {
 			@Override
 			public LedgerHeader prepare(VerifiedVertex vertex) {
 				return LedgerHeader.create(
-					0,
+					vertex.getParentHeader().getLedgerHeader().getEpoch(),
```

Comment:
Perhaps it worth to create specialized versions of LedgerHeader.create() which will cover frequent use cases like creating header from vertex and creating header from proof and validator set. Such methods will serve two purposes: reduce repetitive boilerplate and make code more readable as method name can provide more information than generic 'create'.

Reply:
I agree with the sentiment in certain cases. But in this case, LedgerHeader and VerifiedVertex are data classes of two different modules (consensus + ledger) and should not "know" about each other. Ideally they would never even be used together in the same class. The idea with the current architecture plan is to continually separate these two interfaces which may mean there are some "ugly" code but I think it's fine as long as architecture is correct.